libs/common/include/s25util/macros.h

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#if __cplusplus >= 201703L
+#    define RTTR_NODISCARD [[nodiscard]]
```

Comment:
yea if there is one from boost, lets use that one

Reply:
> It might not be that thorough

A bit of an understatement, now that I had a look. But as long as C++14 isn't set by CMake (I know it isn't *now*), `[[nodiscard]]` works on recent-ish compilers.

Reply:
Ok, then maybe use `[[nodiscard]]` unconditionally which should be the cleanest option and as unknown attributes must be ignored should lead to unintended errors. Or can you see any issue with that?